raspbian/applications/rpi-update.md

```diff
@@ -0,0 +1,37 @@
+# rpi-update
+
+`rpi-update` is a command line application that will update your Raspbian kernel and Videocore firmware to the latest pre-release versions.
+
+**WARNING: Pre-release versions of software are not guaranteed to work. You should not use `rpi-update` on any system unless recommended to do so by a Raspberry Pi engineer. It may leave your system unreliable or even completely broken. It should not be used as part of any regular update process**
+
+The `rpi-update` script is supplied by a third party, `Hexxeh`, and also supported by Raspberry Pi engineers. The script source can be found on Github at https://github.com/Hexxeh/rpi-update
+
+## What it does
+
+`rpi-update` will download the latest pre-release version of the linux kernel, it's matching modules, device tree settings, along with the latest version of the Videocore firmware. It will then install these files to the boot partition of the SD card, overwriting any previous versions. 
+
+All the source data used by `rpi-update` comes from the Github repo https://github.com/Hexxeh/rpi-firmware. This repository simply  contains a subset of the data from the [official firmware repository](https://github.com/raspberrypi/firmware), as not all the data from that repo is required. 
+
+## Running `rpi-update`
+
+`rpi-update` needs to be run as root. Once the update is complete you will need to reboot.
+
+```
+sudo rpi-update
+sudo reboot
+```
+
+It has a number of options, which are documented at the Hexxeh Github repository at https://github.com/Hexxeh/rpi-update
+
+## How to get back to safety
+
+If you have done a `rpi-update` and things are not working as you wish, if your Raspberry Pi is still bootable you can return to the stable release using:
+
+```
+sudo apt-get update
+sudo apt-get install --reinstall raspberrypi-bootloader raspberrypi-kernel
```

Comment:
Are there any userland packages that need to be reinstalled too? @pelwell @popcornmix @XECDesign

Reply:
I don't think so. I believe /opt/vc is included in bootloader package.

Reply:
Hmm, did a bit of poking about on a running Pi 4B (apologies for not doing this earlier) and discovered:
```bash
$ ls /opt/vc
bin  include  lib  src
pi@raspberrypi:~ $ dpkg -S /opt/vc/bin
libraspberrypi-bin: /opt/vc/bin
pi@raspberrypi:~ $ dpkg -S /opt/vc/include
libraspberrypi-dev: /opt/vc/include
pi@raspberrypi:~ $ dpkg -S /opt/vc/lib
libraspberrypi-dev, libraspberrypi0: /opt/vc/lib
pi@raspberrypi:~ $ dpkg -S /opt/vc/src
libraspberrypi-doc: /opt/vc/src
```

`raspberrypi-bootloader` only includes the firmware files in `/boot`
```bash
pi@raspberrypi:~ $ dpkg -L raspberrypi-bootloader 
/.
/boot
/boot/LICENCE.broadcom
/boot/bootcode.bin
/boot/fixup.dat
/boot/fixup4.dat
/boot/fixup4cd.dat
/boot/fixup4db.dat
/boot/fixup4x.dat
/boot/fixup_cd.dat
/boot/fixup_db.dat
/boot/fixup_x.dat
/boot/start.elf
/boot/start4.elf
/boot/start4cd.elf
/boot/start4db.elf
/boot/start4x.elf
/boot/start_cd.elf
/boot/start_db.elf
/boot/start_x.elf
/usr
/usr/share
/usr/share/doc
/usr/share/doc/raspberrypi-bootloader
/usr/share/doc/raspberrypi-bootloader/changelog.Debian.gz
/usr/share/doc/raspberrypi-bootloader/copyright
```

So I guess this implies that `libraspberrypi-bin`, `libraspberrypi-dev`, `libraspberrypi0` and `libraspberrypi-doc` _also_ need to be reinstalled after using `rpi-update` ? IMHO @XECDesign should confirm though.

Reply:
I did notice in another part of the docs that the recovery installs have more in that I wrote here. Does need confirmation.

Reply:
Here https://github.com/raspberrypi/documentation/blob/master/hardware/display/troubleshooting.md#have-you-updated-raspbian

Reply:
I guess that page is another place that should be linking to this new rpi-update page, rather than having duplicated information?

Reply:
...and is it worth adding a link from the new libcamera page to this rpi-update page?